### PR TITLE
Functors task (1) - Poplavskyi

### PR DESCRIPTION
**Pull request by NetLagina**

### CODE REVIEW

File: functors task (1)/student/include/impl.h [hunk 1]
```diff
@@ -8,6 +8,11 @@
  */
 struct BeerOrganizer
 {
+    private:
+        uint currentBeerIndex = 1;
```
Comment (smay1613) [inline functors task (1)/student/include/impl.h:12]: _prefer size_t_

File: functors task (1)/student/include/impl.h [hunk 2]
```diff
@@ -40,9 +47,20 @@ struct MixingPolicy
      * Whiskey + SevenUp = SevenPlusSeven;
      * Others + Others = Oops;
      */
-    static Cocktail mix(/**???*/)
+    static Cocktail mix(AlcoholDrink alcohol, NonAlcoholDrink nonAlcohol)
     {
+        Cocktail cocktail;
+        if (alcohol == AlcoholDrink::Gin && nonAlcohol == NonAlcoholDrink::LimeJuice) {
```
Comment (smay1613) [inline functors task (1)/student/include/impl.h:53]: you can simplify these checks with switch

File: functors task (1)/student/src/impl.cpp [hunk 3]
```diff
@@ -1,2 +1,11 @@
 #include "impl.h"
 
+BeerBrand BeerOrganizer::operator()()
+{
+    if (currentBeerIndex == static_cast<int>(BeerBrand::Max))
+    {
+        currentBeerIndex = 1;
```
Comment (smay1613) [inline functors task (1)/student/src/impl.cpp:7]: _nice, but still magic counters in code_

**Review by smay1613 — CHANGES_REQUESTED** [review]

File: functors task (1)/student/include/impl.h (hunk 1 above)
Comment (NetLagina) [reply functors task (1)/student/include/impl.h]: changed

File: functors task (1)/student/src/impl.cpp (hunk 3 above)
Comment (NetLagina) [reply functors task (1)/student/src/impl.cpp]: ok, changed to enum cast

**Review by smay1613 — COMMENTED** [review]
Nice!

**Pull request closed without merging**